external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -133,32 +133,32 @@ impl ToolchainVersion {
 }
 
 impl Header {
+    /// Read lock file header after verifying that the version of the lock is not newer than the version
+    /// supported by this library.
     pub fn read(lock: &mut impl Read) -> Result<Header> {
         let contents = {
             let mut buf = String::new();
             lock.read_to_string(&mut buf).context("Reading lock file")?;
             buf
         };
-        read_header(&contents)
+        Self::from_str(&contents)
     }
-}
 
-/// Read lock file header after verifying that the version of the lock is not newer than the version
-/// supported by this library.
-#[allow(clippy::ptr_arg)] // Allowed to avoid interface changes.
-pub fn read_header(contents: &String) -> Result<Header> {
-    let Schema { move_: header } =
-        toml::de::from_str::<Schema<Header>>(contents).context("Deserializing lock header")?;
-
-    if header.version > VERSION {
-        bail!(
-            "Lock file format is too new, expected version {} or below, found {}",
-            VERSION,
-            header.version
-        );
-    }
+    #[allow(clippy::ptr_arg)] // Allowed to avoid interface changes.
```

Comment:
I'm assuming the suggestion is from `&String` to `&str` -- this seems pretty harmless, why not adopt it?

Reply:
done!